scrapy/exporters.py

```diff
@@ -214,7 +232,16 @@ def _write_headers_and_set_fields_to_export(self, item):
                     # use fields declared in Item
                     self.fields_to_export = list(item.fields.keys())
             row = list(self._build_row(self.fields_to_export))
-            self.csv_writer.writerow(row)
+            self._writerow(row)
+
+    def _writerow(self, row):
+        self.csv_writer.writerow(row)
+        if six.PY2:
+            data = self.stream.getvalue()
```

Comment:
Python 3 can make use of the TextIOWrapper to write in particular encoding. To write csv files in a specified encoding in Python 2 you need to rely on StringIO, as actually shown in the example in the docs: https://docs.python.org/2/library/csv.html

So basically in python2 the csv writer will write a utf string to StringIO buffer, and then the result will be reencoded to the the specified encoding and written to file